### PR TITLE
Use built in client http caching

### PR DESCRIPTION
**Pull request by asvoboda**

### CODE REVIEW

**Review by bluekeyes — APPROVED** [review]

File: server/server.go
Comment (bluekeyes) [inline server/server.go]: Actually, thinking about this more, I believe we want to use the LRU implementation from start. Bulldozer's set of working objects tends to be new/active PRs and a tail of things that are waiting for review or being updated. Once a PR is merged, the application doesn't care about any state related to that PR, but all those object will stick around in the cache forever.

That means it's guaranteed that the cache will grow without bound, eventually using all memory, unless the application is restarted regularly.

**Review by bluekeyes — CHANGES_REQUESTED** [review]

File: server/config.go
Comment (asvoboda) [inline server/config.go:47]: Do you think we should instead set this config struct in the githubapp library?

File: server/server.go
Comment (asvoboda) [inline server/server.go]: Is there a nice stdlib package that we can use so operators could write something like `50MB` in the config file?

Comment (asvoboda) [reply server/server.go]: Maybe https://godoc.org/github.com/inhies/go-bytesize?

File: server/config.go
Comment (bluekeyes) [reply server/config.go:47]: Let's leave this here for now, with the reasoning that we're not recommending any particular cache implementation in `go-githubapp`. If we end up using it a lot, I could see moving the config up a level and adding something like `WithClientLRUCaching(config)`, but I don't think the config struct alone is valuable without something in the library that uses it.

File: server/server.go
Comment (bluekeyes) [reply server/server.go]: Yeah, nothing in the standard library, but I'd be fine adding either the library you mentioned or one of these:

- https://godoc.org/github.com/docker/go-units
- https://godoc.org/github.com/c2h5oh/datasize

Because `c2h50h/datasize` implements `UnmarshalText`, you _might_ be able to reference it directly in the configuration struct and have the YAML library automatically parse it... not sure if `gopkg.in/yaml.v2` supports that or if it is only `encoding/json`.

File: server/server.go
Comment (bluekeyes) [inline server/server.go]: From https://godoc.org/github.com/die-net/lrucache#New:

>  If maxAge > 0, entries will also be expired after maxAge seconds.

That said, I'm more interested in the size limit than the time limit... I'm not sure if we get any benefit from time-based expiration as long as it can't use unbounded memory.

Comment (asvoboda) [reply server/server.go]: I'd be fine just removing that param 👍

File: server/server.go
Comment (asvoboda) [reply server/server.go]: Oh nice find on the later package, which seems to work as expected.

**Review by bluekeyes — APPROVED** [review]

**Pull request merged**
